contentcuration/contentcuration/static/js/exercise_creation/views.js

```diff
@@ -0,0 +1,562 @@
+            if (!this.setting_model) {
+                this.render_editor();
+            }
+        } else {
+            this.render_content();
+        }
+        this.setting_model = false;
+    },
+
+    render_content: function() {
+        this.$el.html(this.view_template({content: this.model.get(this.edit_key)}));
+    },
+
+    render_editor: function() {
+        this.editor.setHTML(this.view_template({content: this.model.get(this.edit_key)}));
+    },
+
+    activate_editor: function() {
+        this.$el.html(this.edit_template());
+        this.editor = new Quill(this.$(".editor")[0], {
+            modules: {
+                'toolbar': { container: this.$('#toolbar')[0] }
+            },
+            theme: 'snow'
+        });
+        this.render_editor();
+        this.editor.on("text-change", _.debounce(this.save, 500));
+        this.editing = true;
+    },
+
+    deactivate_editor: function() {
+        delete this.editor;
+        this.editing = false;
+        this.render();
+    },
+
+    toggle_editor: function() {
+        if (this.editor) {
+            this.deactivate_editor();
+        } else {
+            this.activate_editor();
+        }
+    },
+
+    save: function(delta, source) {
+        if (typeof source !== "undefined" && source !== "user") {
+            return;
+        }
+        this.setting_model = true;
+        this.model.set(this.edit_key, this.return_markdown());
+    },
+
+    save_and_close: function() {
+        this.save();
+        this.deactivate_editor();
+    },
+
+    return_html: function() {
+        return this.editor.getHTML();
+    },
+
+    return_markdown: function() {
+        var contents = this.editor.getContents();
+        var outputs = [];
+        for (var i = 0; i < contents.ops.length; i++) {
+            var insert = contents.ops[i].insert;
+            var attributes = contents.ops[i].attributes;
+            if (typeof attributes !== "undefined") {
+                _.each(attributes, function(value, key) {
+                    switch (key) {
+                        case "bold":
+                            if (value) {
+                                insert = "**" + insert + "**";
+                            }
+                            break;
+                        case "italic":
+                            if (value) {
+                                insert = "*" + insert + "*";
+                            }
+                            break;
+                        case "image":
+                            if (value && insert == 1) {
+                                insert = "![](" + value + ")";
+                            }
+                            break;
+                    }
+                })
+            }
+            outputs.push(insert);
+        }
+        return outputs.join("");
+    }
+});
+
+
+var set_image_urls_for_export = function(text) {
+    return text.replace(/\!\[[^\]]*\]\((\/media\/)[^\)]*\)/g, "$`web+local://$'");
```

Comment:
Am I correct in thinking you're trying to replace `something![foo](/media/bar/baz)otherthings` with `somethingweb+local://otherthings`?

I get this:
```js
"something![foo](/media/bar/baz)otherthings".replace(/\!\[[^\]]*\]\((\/media\/)[^\)]*\)/g, "$`web+local://$'")
// "somethingsomethingweb+local://otherthingsotherthings"
// notice the duplication
```